include/deal.II/sundials/copy.h

```diff
@@ -0,0 +1,74 @@
+//-----------------------------------------------------------
+//
+//    Copyright (C) 2015 by the deal.II authors
+//
+//    This file is part of the deal.II library.
+//
+//    The deal.II library is free software; you can use it, redistribute
+//    it, and/or modify it under the terms of the GNU Lesser General
+//    Public License as published by the Free Software Foundation; either
+//    version 2.1 of the License, or (at your option) any later version.
+//    The full text of the license can be found in the file LICENSE at
+//    the top level of the deal.II distribution.
+//
+//-----------------------------------------------------------
+
+#ifndef dealii_sundials_copy_h
+#define dealii_sundials_copy_h
+
+#include <deal.II/base/config.h>
+#ifdef DEAL_II_WITH_SUNDIALS
+
+#include <sundials/sundials_nvector.h>
+#include <nvector/nvector_parallel.h>
+#include <nvector/nvector_serial.h>
+
+#include <deal.II/lac/vector.h>
+#include <deal.II/lac/block_vector.h>
+
+#include <deal.II/lac/trilinos_vector.h>
+#include <deal.II/lac/trilinos_parallel_block_vector.h>
+
+#include <deal.II/lac/petsc_parallel_vector.h>
+#include <deal.II/lac/petsc_parallel_block_vector.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+namespace SUNDIALS
+{
+  namespace internal
+  {
+    // The following internal functions are used by SUNDIALS wrappers to copy
+    // to and from deal.II vector types.
+#ifdef DEAL_II_WITH_MPI
+
+#ifdef DEAL_II_WITH_TRILINOS
+    void copy(TrilinosWrappers::MPI::Vector &dst, const N_Vector &src);
```

Comment:
You choose the `vmult` convention of saying `copy(dst,src)` (which mirrors `dst = src`). But that's not what `std::copy` does, for example, nor what most of the rest of the deal.II functions do. Thoughts?

Reply:
I kept this in an internal namespace, because ideally we should get rid of all these in favour of `ReadWriteVector` and `VectorSpaceVector` objects. All copy functions should disappear.